core/finality-grandpa/src/environment.rs

```diff
@@ -26,8 +26,9 @@ use tokio_timer::Delay;
 use parking_lot::RwLock;
 
 use client::{
-	backend::Backend, BlockchainEvents, CallExecutor, Client, error::Error as ClientError,
-	utils::is_descendent_of,
+	backend::Backend, apply_aux, BlockchainEvents, CallExecutor,
```

Comment:
How does `apply_aux` work now?

Reply:
[effectivly the same](https://github.com/paritytech/substrate/pull/2960/files#diff-e1db240b5937ee7558375dc760bc09f4R1861) as before. It was just a [move from client(https://github.com/paritytech/substrate/pull/2960/files#diff-e1db240b5937ee7558375dc760bc09f4L1232), which it didn't use or need, as its own function, breaking the unneeded dependency of `client`. Code hasn't actually changed.